internal/core/commands/sync_vin_numbers.go

```diff
@@ -0,0 +1,168 @@
+package commands
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/DIMO-Network/device-definitions-api/internal/core/common"
+	"github.com/DIMO-Network/device-definitions-api/internal/core/services"
+	"github.com/DIMO-Network/device-definitions-api/internal/infrastructure/db/models"
+	"github.com/DIMO-Network/device-definitions-api/internal/infrastructure/db/repositories"
+	"github.com/DIMO-Network/shared"
+	"github.com/DIMO-Network/shared/db"
+	"github.com/TheFellow/go-mediator/mediator"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"github.com/segmentio/ksuid"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+)
+
+type SyncVinNumbersCommand struct {
+	VINNumbers []string
+}
+
+type SyncVinNumbersCommandResult struct {
+	Status bool
+}
+
+func (*SyncVinNumbersCommand) Key() string { return "SyncVinNumbersCommand" }
+
+type SyncVinNumbersCommandHandler struct {
+	dbs                func() *db.ReaderWriter
+	vinDecodingService services.VINDecodingService
+	logger             *zerolog.Logger
+	repository         repositories.DeviceDefinitionRepository
+	vinRepository      repositories.VINRepository
+}
+
+func NewSyncVinNumbersCommand(dbs func() *db.ReaderWriter,
+	vinDecodingService services.VINDecodingService,
+	repository repositories.DeviceDefinitionRepository,
+	vinRepository repositories.VINRepository,
+	logger *zerolog.Logger) SyncVinNumbersCommandHandler {
+	return SyncVinNumbersCommandHandler{
+		dbs:                dbs,
+		vinDecodingService: vinDecodingService,
+		logger:             logger,
+		repository:         repository,
+		vinRepository:      vinRepository,
+	}
+}
+
+func (dc SyncVinNumbersCommandHandler) Handle(ctx context.Context, query mediator.Message) (interface{}, error) {
+
+	command := query.(*SyncVinNumbersCommand)
+	for _, vinNumber := range command.VINNumbers {
+		if len(vinNumber) != 17 {
+			dc.logger.Warn().Str("vin", vinNumber).Msgf("invalid vin %s", vinNumber)
+			continue
+		}
+
+		vin := shared.VIN(vinNumber)
+		year := int32(vin.Year()) // needs to be updated for newer years
+		wmi := vin.Wmi()
+
+		localLog := dc.logger.With().
+			Str("vin", vin.String()).
+			Str("handler", query.Key()).
+			Str("year", fmt.Sprintf("%d", year)).
+			Logger()
+
+		vinDecodeNumber, err := models.FindVinNumber(ctx, dc.dbs().Reader, vin.String())
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				localLog.Debug().Msg("No rows to decode vin numbers from db")
+			}
+		}
+
+		if vinDecodeNumber == nil {
```

Comment:
`if vinDecodeNumber != nil ` we should return the object and short circuit the rest of the operation since we already have seen it before.

Reply:
oh i see it just continues